File: resources/execute/model/data_result.py
from flask import jsonify
from model.config.error_info import ErrorInfo
# res.send({ status: 'Fail', message: error.message, data: null })
class DataResult():
    def __init__(self):
        self.status = 200
        self.message = 'success'
        self.data = None

    def to_dict(self):
        return {
            'code': self.status,
            'msg': self.message,
            'data': self.data
        }
    
    @staticmethod
    def ok(data='success'):
        result = DataResult()
        result.data = data
        response = jsonify(result.to_dict())
        # response.status_code = 200
        # response.headers.add('Content-Type', 'application/json')
        return response
    @staticmethod
    def fail(msg, code=ErrorInfo.code_err_default):
        result = DataResult()
        result.data = None
        result.status = code
        result.message = msg
        response = jsonify(result.to_dict())
        response.status_code = 200
        response.headers.add('Content-Type', 'application/json')
        return response

    @staticmethod
    def reply_exception(msg, code=ErrorInfo.code_err_default):
        result = DataResult()
        result.data = None
        result.status = code
        result.message = msg
        result.data = msg
        response = jsonify(result.to_dict())
        response.status_code = 200
        response.headers.add('Content-Type', 'application/json')
        return response

File: src/model/DataResult.js
// src/models/dataResult.js
export default class DataResult {
    constructor(code, data, msg) {
      this.code = code; // 状态码
      this.data = data; // 返回的数据
      this.msg = msg;   // 提示信息
    }
  
    // 成功的响应
    static success(data, msg = 'success') {
      return new DataResult(200, data, msg);
    }
  
    // 失败的响应
    static fail(msg, code = 400) {
      return new DataResult(code, null, msg);
    }
  
    // 错误的响应
    static error(msg, code = 500) {
      return new DataResult(code, null, msg);
    }
  
    // 判断是否成功
    isSuccess() {
      return this.code === 200;
    }
  
    // 判断是否失败
    isFail() {
      return this.code !== 200;
    }
  }